libs/SalesforceSDK/src/com/salesforce/androidsdk/security/PasscodeManager.java

```diff
@@ -397,9 +397,7 @@ public boolean isLocked() {
      * @param ctx
      */
     public void lock(Context ctx) {
-        locked = true;
```

Comment:
the first problem i found in the passcode change flow is that these two lines were happening in 'lock()' but not in 'showLockActivity' which is what the passcode change flow calls. I've just moved them to happen in that method.